iidm/iidm-mergingview/src/test/java/com/powsybl/iidm/mergingview/BusAdapterTest.java

```diff
@@ -72,12 +73,53 @@ public void testSetterGetter() {
     }
 
     @Test
-    public void testComponentSetterGetter() {
+    public void testSynchronousComponentSetterGetter() {
+        MergingView view = MergingView.create("testSynchronousComponentSetterGetter", "iidm");
+        view.merge(HvdcTestNetwork.createVsc());
+        HvdcLine l = view.getHvdcLine("L");
+        assertNotNull(l);
+        VscConverterStation cs1 = view.getVscConverterStation("C1");
+        VscConverterStation cs2 = view.getVscConverterStation("C2");
+        cs1.setLossFactor(0.022f);
+        cs1.setVoltageSetpoint(406.0);
+        cs2.setReactivePowerSetpoint(124.0);
+        cs2.setVoltageSetpoint(405);
+        cs2.setVoltageRegulatorOn(true);
+        Bus bus1 = l.getConverterStation1().getTerminal().getBusView().getBus();
+        Bus bus2 = l.getConverterStation2().getTerminal().getBusView().getBus();
+        assertTrue(bus1.isInMainConnectedComponent());
+        assertTrue(bus2.isInMainConnectedComponent());
+        assertTrue(bus1.isInMainSynchronousComponent());
+        assertFalse(bus2.isInMainSynchronousComponent());
+        int num1 = bus1.getSynchronousComponent().getNum();
+        int num2 = bus2.getSynchronousComponent().getNum();
+        assertNotEquals(num1, num2);
+    }
+
+    @Test
+    public void testConnectedComponentSetterGetter() {
+        MergingView view = MergingView.create("testConnectedComponentSetterGetter", "iidm");
+        view.merge(NetworkTest1Factory.create());
+        VoltageLevel voltageLevel1 = view.getVoltageLevel("voltageLevel1");
+        VoltageLevel.NodeBreakerView topology1 = voltageLevel1.getNodeBreakerView();
+        BusbarSection voltageLevel1BusbarSection1 = topology1.getBusbarSection("voltageLevel1BusbarSection1");
+        BusbarSection voltageLevel1BusbarSection2 = topology1.getBusbarSection("voltageLevel1BusbarSection2");
+        Load load1 = view.getLoad("load1");
+        Generator generator1 = view.getGenerator("generator1");
+        Bus busCalc1 = voltageLevel1BusbarSection1.getTerminal().getBusBreakerView().getBus();
+        Bus busCalc2 = voltageLevel1BusbarSection2.getTerminal().getBusBreakerView().getBus();
+        assertSame(busCalc1, load1.getTerminal().getBusBreakerView().getBus());
+        assertSame(busCalc2, generator1.getTerminal().getBusBreakerView().getBus());
+
+        assertTrue(busCalc1.isInMainConnectedComponent());
+        assertTrue(busCalc2.isInMainConnectedComponent());
+        assertEquals(0, busCalc1.getConnectedComponent().getNum());
+        assertEquals(0, busCalc2.getConnectedComponent().getNum());
+    }
+
+    @Test
+    public void testComponentVisitor() {
```

Comment:
To be sure that everything is working, you have to merge two networks:
- the two network should have 2 connected components with two different sizes
- once merged, the two smallest connected components should be the main connected component.

In that case, from a CGM point of view, the main component is not the same than from the IGM point of view. This is exacly why I think we should add components in BusAdapter and TerminalAdapter.

But, if the topology changes, in the underlying network, you have to update both connected components (IGM and CGM).